internal/test/e2e/tinkerbellHardwareQueue.go

```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+)
+
+// Default timeout for Tink tests to poll for hardware.
+const hwPollingTimeout = 120 * time.Minute
+
+type hardwareQueue struct {
+	hws []*api.Hardware
+	mu  sync.Mutex
+}
+
+func (hwQu *hardwareQueue) reserveHardware(count int) ([]*api.Hardware, error) {
```

Comment:
Callers have no way of terminating this process. Here's an alternative:

```go
func (q *hardwareQueue) ReserveHardware(ctx context.Context, count int) []*api.Hardware, error) {
  ctx, cancel := context.WithTimeout(ctx, hwPollingTimeout)
  defer cancel()

  ticker := time.NewTicker(1 * time.Minute)
  defer ticker.Stop()

  for {
    select{
    case <-ctx.Done():
      return nil, ctx.Err()

    case <-ticker.C:
      // No timeout logic, only reservation logic.
    }
  }
}
```

While the ticker approach is OK because the ticks are 1m apart, an alternative approach is to use conditions with a notification being published every time hardware is enqueued. That way, there's no need to wait up-to 1m before returning reserved hardware.

Reply:
At first I thought of having a context but didn't see it adding any additional value for two reasons. One we are just dealing with a hard timeout (in which contexts can be useful if passed around to multiple functions)  and not any additional conditions. The entire file doesn't have context in any of the function signatures, so it doesn't add any interoperability. Also, the existing function returns the hardware as soon as it is reserved and doesn't wait up to a minute to return it! Since we are not making a request to an external resource like say a DB or server didn't find extra value in introducing contexts here!